lago/plugins/vm.py

```diff
@@ -411,22 +423,39 @@ def copy_to(self, local_path, remote_path, recursive=True):
         with LogTask(
             'Copy %s to %s:%s' % (local_path, self.name(), remote_path),
         ):
-            with self._scp() as scp:
-                scp.put(
-                    files=local_path,
-                    remote_path=remote_path,
-                    recursive=recursive,
-                )
+            try:
+                with self._scp() as scp:
+                    scp.put(
+                        files=local_path,
+                        remote_path=remote_path,
+                        recursive=recursive,
+                    )
+            except OSError:
+                raise LagoCopyFilesToVMError(local_path)
 
     def copy_from(
         self, remote_path, local_path, recursive=True, propagate_fail=True
     ):
-        with self._scp(propagate_fail=propagate_fail) as scp:
-            scp.get(
-                recursive=recursive,
-                remote_path=remote_path,
-                local_path=local_path,
-            )
+        with LogTask(
+            'Copy from %s:%s to %s' % (self.name(), remote_path, local_path),
+        ):
+            try:
+                with self._scp(propagate_fail=propagate_fail) as scp:
+                    scp.get(
+                        recursive=recursive,
+                        remote_path=remote_path,
+                        local_path=local_path,
+                    )
+            except SCPException as err:
+                err_substr = ': No such file or directory'
+                if len(err.args) > 0 and isinstance(
+                    err.args[0], basestring
+                ) and err_substr in err.args[0]:
+                    if propagate_fail:
```

Comment:
`propagate_fail` is a variable should be used only by the logger.
You can see how it's being used in `lago.log_utils:604`.

In our case, I think that we can just raise the exception.